From 4737dfd713700ad70bbccf94dc7801750049fc53 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Tue, 11 Feb 2025 13:49:48 +0000
Subject: [PATCH] dt-bindings: pci: pcie-brcmstb: add BCM2712-specific
 properties

There is configurable priority forwarding hardware in this variant of the
Root Complex controller. Add optional properties to configure FIFO
backpressure or Vendor-Defined Message priority forwarding.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 .../bindings/pci/brcm,stb-pcie.yaml           | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

--- a/Documentation/devicetree/bindings/pci/brcm,stb-pcie.yaml
+++ b/Documentation/devicetree/bindings/pci/brcm,stb-pcie.yaml
@@ -84,6 +84,14 @@ properties:
     $ref: /schemas/types.yaml#/definitions/string
     enum: [ safe, no-l1ss, default ]
 
+  brcm,fifo-qos-map:
+    description: Array of u8 elements which assigns every per-TC FIFOs
+      an AXI priority based on fullness quartile (backpressure signalling).
+      Mutually exclusive with vdm-qos-map.
+    $ref: /schemas/types.yaml#/definitions/uint8-array
+    minItems: 4
+    maxItems: 4
+
   brcm,scb-sizes:
     description: u64 giving the 64bit PCIe memory
       viewport size of a memory controller.  There may be up to
@@ -97,6 +105,15 @@ properties:
       minItems: 1
       maxItems: 3
 
+  brcm,vdm-qos-map:
+    description: Array of u8 elements which assigns each per-TC FIFO
+      a base AXI priority with automatic elevation depending on
+      Vendor Messages from the EP - specifically, RP1.
+      Mutually exclusive with fifo-qos-map.
+    $ref: /schemas/types.yaml#/definitions/uint8-array
+    minItems: 8
+    maxItems: 8
+
   resets:
     minItems: 1
     maxItems: 3
@@ -187,6 +204,17 @@ allOf:
         - resets
         - reset-names
 
+  - if:
+      not:
+        properties:
+          compatible:
+            contains:
+              const: brcm,bcm2712-pcie
+    then:
+      properties:
+        brcm,fifo-qos-map: false
+        brcm,vdm-qos-map: false
+
 unevaluatedProperties: false
 
 examples:
